src/packing_queue/tables/PendingTable.jsx

```diff
@@ -0,0 +1,483 @@
+              rowsPerPage={numRowsPerPage}
+              onRowsPerPageChange={(event) => {
+                const pageValue = parseInt(event.target.value, 10);
+                onPageSizeChange(
+                  pageValue,
+                  page,
+                  queueData.length,
+                  setPage,
+                  setNumRowsPerPage
+                );
+              }}
+              onPageChange={handlePageChange}
+              page={page}
+              sx={{ border: "0px" }}
+            />
+          </tr>
+        </tbody>
+      </table>
+    );
+  }, [page, queueData.length, numRowsPerPage, setNumRowsPerPage]);
+
+  return (
+    <div className={classes.root}>
+      <DataGrid
+        sx={{
+          border: "none",
+          height: `calc(100vh - ${PACKING_SLIP_BOTTOM_MARGIN} - ${PACKING_SLIP_TOP_MARGIN} - ${NAV_BAR_HEIGHT} - 5rem)`,
+          minHeight: "20rem",
+        }}
+        className={classes.table}
+        rows={
+          isLoading
+            ? []
+            : filteredPending.slice(
+                page * numRowsPerPage,
+                page * numRowsPerPage + numRowsPerPage
+              )
+        }
+        columns={columns}
+        pageSize={numRowsPerPage}
+        rowsPerPageOptions={PAGINATION_SIZING_OPTIONS}
+        columnBuffer={0}
+        disableColumnSelector
+        disableDensitySelector
+        checkboxSelection={false}
+        disableSelectionOnClick={true}
+        loading={isLoading}
+        sortModel={sortModel}
+        onSortModelChange={(model) => {
+          setSortModel(model);
+        }}
+        components={{
+          LoadingOverlay: () => <PackShipProgress />,
+          Footer: () => generateTablePagination(),
+        }}
+        componentsProps={{
+          row: {
+            onContextMenu: handleContextMenu,
+          },
+        }}
+      />
+
+      <ContextMenu contextMenu={contextMenu} setContextMenu={setContextMenu}>
+        {historyRowMenuOptions}
+      </ContextMenu>
+      <EditPackingSlipDialog
+        isOpen={isEditPackingSlipOpen.open}
+        viewOnly={isEditPackingSlipOpen.viewOnly}
+        onClose={onPackingSlipClose}
+        onSubmit={onPackingSlipSubmit}
+        onDestinationChange={onDestinationChange}
+        packingSlipData={selectedRow}
+        onAdd={onHistoryPackingSlipAdd}
+        onNewPartRowChange={onNewPartRowChange}
+        onPackQtyChange={onPackQtyChange}
+        onDelete={(params) => {
+          setConfirmDeleteDialogOpen(true);
+          setItemToDelete(params.row);
+        }}
+      />
+      <ConfirmDialog
+        title="Are You Sure You Want To Delete This?"
+        open={confirmDeleteDialogOpen}
+        setOpen={setConfirmDeleteDialogOpen}
+        onConfirm={onItemDelete}
+      />
+      <ConfirmDialog
+        title="Do You Want To Delete This?"
+        open={deleteDialog}
+        setOpen={setDeleteDialog}
+        onConfirm={deletePackingSlip}
+      />
+    </div>
+  );
+};
```

Comment:
HOC reminder for the context menu. check out #136

Reply:
Pulled this out

Reply:
Made HOC